### PR TITLE
Jetty 12 - Review BOMs

### LINKED ISSUE #9467: Jetty 12 - Review BOMs
**Jetty version(s)**
12+

**Description**
There currently are mistakes in the BOM artifact/pom.xml referencing old artifacts ID that have been moved to eeN.
Also, jetty-bom/ and jetty-core/jetty-bom have the same groupId/artifactId, why do we need both?
Etc.

### PR DESCRIPTION
**Pull request by joakime**
Fixes #9467

### CODE REVIEW

File: jetty-core/jetty-bom/pom.xml [hunk 1]
```diff
@@ -85,221 +85,208 @@
...
-        <artifactId>jetty-quic-common</artifactId>
+        <groupId>org.eclipse.jetty.http2</groupId>
+        <artifactId>jetty-http2-server</artifactId>
         <version>12.0.0-SNAPSHOT</version>
       </dependency>
       <dependency>
-        <groupId>org.eclipse.jetty.quic</groupId>
-        <artifactId>jetty-quic-quiche-common</artifactId>
+        <groupId>org.eclipse.jetty.http2</groupId>
+        <artifactId>jetty-http2-tests</artifactId>
```
Comment (sbordet) [inline jetty-core/jetty-bom/pom.xml]: I don't think this is deployed, as it's only tests. I'd remove it.

File: jetty-ee10/jetty-ee10-bom/pom.xml [hunk 2]
```diff
@@ -105,6 +110,26 @@
...
+        <version>12.0.0-SNAPSHOT</version>
+      </dependency>
+      <dependency>
+        <groupId>org.eclipse.jetty.ee10.osgi</groupId>
+        <artifactId>jetty-ee10-osgi-boot-jsp</artifactId>
+        <version>12.0.0-SNAPSHOT</version>
+      </dependency>
+      <dependency>
+        <groupId>org.eclipse.jetty.ee10.osgi</groupId>
+        <artifactId>test-jetty-ee10-osgi</artifactId>
```
Comment (sbordet) [inline jetty-ee10/jetty-ee10-bom/pom.xml]: Test artifact, I'd remove it.

File: jetty-ee9/jetty-ee9-bom/pom.xml [hunk 3]
```diff
@@ -128,7 +125,21 @@
...
+        <version>12.0.0-SNAPSHOT</version>
+      </dependency>
+      <dependency>
+        <groupId>org.eclipse.jetty.ee9.osgi</groupId>
+        <artifactId>jetty-ee9-osgi-boot-jsp</artifactId>
+        <version>12.0.0-SNAPSHOT</version>
+      </dependency>
+      <dependency>
+        <groupId>org.eclipse.jetty.ee9.osgi</groupId>
+        <artifactId>test-jetty-ee9-osgi</artifactId>
```
Comment (sbordet) [inline jetty-ee9/jetty-ee9-bom/pom.xml]: Test artifact? I'd remove it.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-core/jetty-bom/pom.xml (hunk 1 above)
Comment (joakime) [reply jetty-core/jetty-bom/pom.xml]: It is being deployed.
I'll also flag it as `maven.deploy.skip=true`

File: jetty-ee10/jetty-ee10-bom/pom.xml (hunk 2 above)
Comment (joakime) [reply jetty-ee10/jetty-ee10-bom/pom.xml]: I'll also flag it as `maven.deploy.skip=true`

File: jetty-ee9/jetty-ee9-bom/pom.xml (hunk 3 above)
Comment (joakime) [reply jetty-ee9/jetty-ee9-bom/pom.xml]: I'll also flag it as `maven.deploy.skip=true`

**Review by sbordet — APPROVED** [review]

**Review by olamy — APPROVED** [review]

**Pull request merged**